### ISSUE TITLE
Consider updating docs/readme to addNetworkInterceptor() instead of addInterceptor()

### ISSUE THREAD

**Issue by ColtonIdle** [ChuckerTeam/chucker#759]
## :warning: Is your feature request related to a problem? Please describe
I recently had an issue where I couldn't see my headers in the request tab. I use chucker as an on-device replacement for chalres proxy and so not showing okhttp headers by default (like the docs show) is sub-par imo. So I'm basically turning this comment by @MiSikora into a proper feature request. https://github.com/ChuckerTeam/chucker/issues/758#issuecomment-1007020678

## :bulb: Describe the solution you'd like
Update docs from

```
// Don't forget to plug the ChuckerInterceptor inside the OkHttpClient
val client = OkHttpClient.Builder()
        .addInterceptor(chuckerInterceptor)
        .build()
```

to

```
// Don't forget to plug the ChuckerInterceptor inside the OkHttpClient
val client = OkHttpClient.Builder()
        .addNetworkInterceptor(chuckerInterceptor) // use network interceptor if you want to show all network traffic (like a proxy would)
        // or
        .addInterceptor(chuckerInterceptor) // use an application interceptor if you want to show... idk why you would want an app interceptor honestly. 🤷‍♀️ )
        .build()
```

## :page_facing_up: Additional context
As per the okhttp docs... Network Interceptors "Observe the data just as it will be transmitted over the network." so to me...
chucker is a library that is made to replace a proxy
a proxy is used to observes data over the network
why recommend/guide users towards using an application interceptor?

Anyway. I just hope some minor changes to docs will help someone make an informed decision when adding the library, vs just following the docs for adding an app interceptor.

**Comment by MiSikora** [comment]
Could you make a PR for this?

> why recommend/guide users towards using an application interceptor?

I don't think it is a recommendation. Just the way things were documented. I personally never use Chucker as an application interceptor, but I think there are some advantages. For example, you don't pollute calls preview with redirect responses.

**Comment by ColtonIdle** [comment]
I didn't open a PR because I didn't know if there were some advantages to app vs network (you highlighted a good one though "not polluting chucker with redirect responses"

I can try to look up some more pros and cons to try to compare them at some point in the future.

**Comment by ColtonIdle** [comment]
Yeah. To me it's less about that section, but more about the fact that the docs say this:

```
// Don't forget to plug the ChuckerInterceptor inside the OkHttpClient
val client = OkHttpClient.Builder()
        .addInterceptor(chuckerInterceptor)
        .build()
```

You gotta admit that to people that aren't familiar to interceptors that this will go right over their head. And I'd make the same argument about headers. I didn't even realize I had missing headers in the request tabs because I mostly look at the response tab, but today I had to figure out why a request wasn't working and I was surprised that my app for like over a year was missing headers in the request tab section. lol

**Comment by MiSikora** [comment]
> I didn't even realize I had missing headers in the request tabs because I mostly look at the response tab

When Chucker is added as an application interceptor, some of the response's headers might also be missing.

https://github.com/square/okhttp/blob/b8ebe99bb4b6b516b4a2e01ef103fc536a0cf502/okhttp/src/jvmMain/kotlin/okhttp3/internal/http/BridgeInterceptor.kt#L95-L99

https://github.com/square/okhttp/blob/b8ebe99bb4b6b516b4a2e01ef103fc536a0cf502/okhttp-brotli/src/main/kotlin/okhttp3/brotli/internal/brotli.kt#L41-L45

**Comment by ColtonIdle** [comment]
On kotlinglang slack jakewharton answered a question of mine and said:

> Network interceptor sees all, especially if it's last in the chain

With that said, I'm definitely moving to a network interceptor thats last in the chain since I want chucker to act as a complete replacement of charles proxy.

**Comment by trevjonez** [comment]
I make heavy use of interceptors to do dynamic request modifications in every app I build in order to keep my retrofit services high level. Things like building the base url to use based on annotations in order to target different environments.

Looking at the [okhttp docs](https://square.github.io/okhttp/interceptors/#choosing-between-application-and-network-interceptors) there is one item that really stands out to me.
> Observe the data just as it will be transmitted over the network.

This also means the logging interceptors need to be the last ones added so they are the last to run.

for the concern of
> not polluting chucker with redirect responses

Perhaps adding an option to collapse redirects to `ChuckerInterceptor.Builder` would be sufficient? 
Chucker could tag the request so that it is easier to track what redirect came from what original request (assuming there isn't some other clear mechanism to track that)?

**Comment by ColtonIdle** [comment]
I found out today that charles also shows redirects and so im okay with them because i want chucker to replaces charles. 😄

**Comment by ravenfeld** [comment]
Thanks for your issue. I spent a long time trying to figure out why my cache wasn't working, but it turns out it was working — it was just Chucker's application-level interceptor that was misleading me

**Comment by ColtonIdle** [comment]
Nice to know this was finally added! Woo

I'm saving two other notes here for the future since I got responses from both jake wharton and jesse wilson on some "weird" chucker end result (which is just because of how application vs network interceptors work)

> I'm using [Chucker](https://github.com/ChuckerTeam/chucker) as a networkInterceptor to see network requests. I am seeing 0 network requests. Moved to using plain ol' interceptor and I can see that I do in fact make a network request (like I thought) but it failed due to SSLHandshakeException.

Jake said
Because if TLS fails there is no connection over which to send the request

> Sometimes I see !!! in chucker since i switched to using a network interceptor. Can someone help me debug why that is?

Jesse wilson said (after a bunchhhhh of debugging)
The server is closing the connection, the client is attempting to reuse
